### PR TITLE
py-ephem: a new python package.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-ephem/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    performing high-precision astronomy computations."""
+
+    homepage = "https://rhodesmill.org/pyephem/"
+    url      = "https://pypi.python.org/packages/source/e/ephem/ephem-3.7.6.0.tar.gz"
+
+    version('3.7.6.0', sha256='7a4c82b1def2893e02aec0394f108d24adb17bd7b0ca6f4bc78eb7120c0212ac')
+
+    variant('docs', default=False, description='build documentation')
+
+    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ephem/package.py]: The package claims to support 2.6 and 3.2, but it's not enforced anywhere. I would just remove this line.

File: var/spack/repos/builtin/packages/py-ephem/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+class PyEphem(PythonPackage):
+    """PyEphem provides an ephem Python package for
+    performing high-precision astronomy computations."""
+
+    homepage = "https://rhodesmill.org/pyephem/"
+    url      = "https://pypi.python.org/packages/source/e/ephem/ephem-3.7.6.0.tar.gz"
+
+    version('3.7.6.0', sha256='7a4c82b1def2893e02aec0394f108d24adb17bd7b0ca6f4bc78eb7120c0212ac')
+
+    variant('docs', default=False, description='build documentation')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ephem/package.py]: This variant doesn't do anything

File: var/spack/repos/builtin/packages/py-ephem/package.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ephem/package.py]: Setuptools is not used by this project in any way, which is why you had to hack the args below, remove both

File: var/spack/repos/builtin/packages/py-ephem/package.py (hunk 1 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/py-ephem/package.py]: I've removed setuptools dependency.

Comment (weijianwen) [reply var/spack/repos/builtin/packages/py-ephem/package.py]: Requirements for python versions are also removed.

File: var/spack/repos/builtin/packages/py-ephem/package.py (hunk 2 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/py-ephem/package.py]: docs variant has been removed.

File: var/spack/repos/builtin/packages/py-ephem/package.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+class PyEphem(PythonPackage):
+    """PyEphem provides an ephem Python package for
+    performing high-precision astronomy computations."""
+
+    homepage = "https://rhodesmill.org/pyephem/"
+    url      = "https://github.com/brandon-rhodes/pyephem/archive/v3.7.7.1.tar.gz"
+
+    version('3.7.7.1', sha256='d9d05d85c0d38a79169acaef25964ac9df2d808f0d833354545b9ef681ff584d')
+
+    depends_on('python', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ephem/package.py:18]: This isn't needed, it's already in the PythonPackage base class

File: var/spack/repos/builtin/packages/py-ephem/package.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    # Prevent passing --single-version-externally-managed to
+    # setup.py, which it does not support.
+    # code copied from scons/package.py
+    def install_args(self, spec, prefix):
+        return ['--prefix={0}'.format(prefix), '--root=/']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ephem/package.py:24]: Same here, you can remove this now

File: var/spack/repos/builtin/packages/py-ephem/package.py (hunk 4 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/py-ephem/package.py:18]: Fixed.

File: var/spack/repos/builtin/packages/py-ephem/package.py (hunk 5 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/py-ephem/package.py:24]: Fixed.

**Comment by adamjstewart** [comment]
Flake8:
```
var/spack/repos/builtin/packages/py-ephem/package.py:17: [W391] blank line at end of file
```

**Comment by weijianwen** [comment]
Remove the empty line.

**Pull request merged**